### PR TITLE
Deprecate XTOS on imx8/imx8x platforms

### PR DESCRIPTION
**Pull request by LaurentiuM1234**
To prepare for XTOS deprecation and switch to Zephyr native drivers on i.MX8QM/i.MX8QXP, remove the ability to build SOF with XTOS on said platforms. Also, remove all XTOS builds/tests for these platforms from CI.

### CODE REVIEW

**Comment by dbaluta** [comment]
Thanks @LaurentiuM1234 I'm discussing now with our Yocto team in order to define the strategy of removing XTOS support. Will merge this once we have an agreement.

**Comment by LaurentiuM1234** [comment]
> Is there some public, very short document comparing "QM" versus "QXP" and a small table showing which "imx8?" belongs to what? It's pretty cryptic otherwise.

Sorry for the confusion. Not sure we have that. Maybe @dbaluta can help here. For reference:

1. imx8 - QM
2. imx8x - QXP

The "imx8" and "imx8x" naming stems from the name of the family.


> Can this wait until Zephyr updates are unblocked? I mean:
> 
> * [[BUG] Zephyr baseline update blocked #8818](https://github.com/thesofproject/sof/issues/8818)

The transition is blocked by the same issue anyways so this can wait as well

**Comment by iuliana-prodan** [comment]
> Is there some public, very short document comparing "QM" versus "QXP" and a small table showing which "imx8?" belongs to what? It's pretty cryptic otherwise.
> 
https://www.nxp.com/products/processors-and-microcontrollers/arm-processors/i-mx-applications-processors/i-mx-8-applications-processors:IMX8-SERIES
From this link you can go for [i.MX 8](https://www.nxp.com/products/processors-and-microcontrollers/arm-processors/i-mx-applications-processors/i-mx-8-applications-processors/i-mx-8-family-arm-cortex-a53-cortex-a72-virtualization-vision-3d-graphics-4k-video:i.MX8) or [i.MX 8X](https://www.nxp.com/products/processors-and-microcontrollers/arm-processors/i-mx-applications-processors/i-mx-8-applications-processors/i-mx-8x-family-arm-cortex-a35-3d-graphics-4k-video-dsp-error-correcting-code-on-ddr:i.MX8X) or [i.MX 8ULP](https://www.nxp.com/products/processors-and-microcontrollers/arm-processors/i-mx-applications-processors/i-mx-8-applications-processors/i-mx-8ulp-applications-processor-family:i.MX8ULP) or 
[i.MX 8M Plus](https://www.nxp.com/products/processors-and-microcontrollers/arm-processors/i-mx-applications-processors/i-mx-8-applications-processors/i-mx-8m-plus-arm-cortex-a53-machine-learning-vision-multimedia-and-industrial-iot:IMX8MPLUS) - in each of this links look for "_Part numbers include:_ "

**Comment by dbaluta** [comment]
> Is there some public, very short document comparing "QM" versus "QXP" and a small table showing which "imx8?" belongs to what? It's pretty cryptic otherwise.

|Family              |    Instance name | Application processor cores | DSP |
| --------------------| ---------------------  |--------------------------------------|-----------|
| i.MX8               | i.MX8QM            | 2 x Cortex A72 + 4 x Cortex A53  | HIFI4@666Mhz |
| i.MX8X             | i.MX8QXP          |4 x Cortex A35                               | HIFI4@640MHhz|
|i.MX8M             | i.MX8MPlus        |4 x Cortex A53                               | HIFI4@800MHz |
|i.MXULP          | i.MX8ULP            |2 X Cortex A35                               | HIFI4@528Mhz|

image

File: scripts/test-repro-build.sh
Comment (lgirdwood) [reply scripts/test-repro-build.sh:42]: This can be done incrementally by AMD or MTK, not a blocker for this today.

**Review by lgirdwood — APPROVED** [review]

**Comment by marc-hb** [comment]
Thanks for the table, it is useful. I initially wondered whether it would be useful to add it as a comment somewhere or in sof-docs but I suspect there's a much simpler solution... would it be possible to very simply avoid "QM" and "QX" in commit messages and sources completely and just stick to "imx8" and "imx8m" as it has been done until now? As far as SOF is concerned it looks like a 1:1 mapping so far so it does not look like information would be lost if dropping "QM" and "QX" completely?

CI compiles multiple configurations in many different ways so the time wasted to find this sort of mapping can add up quickly; it's not negligible. Also, fixing issues in other vendors' configurations is always quite stressful so every little helps.

File: scripts/test-repro-build.sh
Comment (LaurentiuM1234) [reply scripts/test-repro-build.sh]: Changed to AMD's rmb. We're going to have to change to other XTOS platforms at some point anyways so might as well do it now if the change is very simple

**Comment by LaurentiuM1234** [comment]
**V2 updates**

1. Fixed commit messages to use imx8/imx8x instead of QM/QXP.
2. Switched to AMD's rmb in reproducible builds test.

**Review by dbaluta — APPROVED** [review]

**Review by kv2019i — APPROVED** [review]

**Pull request merged**
